src/network.hpp

```diff
@@ -0,0 +1,119 @@
+#pragma once
+
+// opencv
+#include <opencv2/core.hpp>
+
+// spdlog
+#include <spdlog/spdlog.h>
+
+// std
+#include <algorithm>
+#include <random>
+#include <vector>
+
+// tuple holding information about input of a training data and expected output
+// out of it
+struct Sample {
+public:
+  Sample() = default;
+  Sample(const std::vector<double> &input,
+         const std::vector<double> &expected_output)
+      : input(input), expectedOutput(expected_output) {}
+
+public:
+  std::vector<double> input;
+  std::vector<double> expectedOutput;
+};
+
+// Represents simple feed forward Neural Network.
+// As a cost function we use cross entropy and neuron type is sigmoid neurons.
+// It supports learning by SGD and simple L2 regularization technique.
```

Comment:
Would we want to use a library for creating neural networks in the future? For easy access to more advanced techniques and better performance.

Reply:
Yeah, if this Network would be too slow or too clancky to update we can use a library for creating neural networks.